sky/tools/create_macos_framework.py

```diff
@@ -102,40 +103,57 @@ def zip_framework(dst):
   final_dst_path = os.path.join(dst, 'FlutterMacOS.framework.zip')
   shutil.move(final_src_path, final_dst_path)
 
-  zip_xcframework_archive(dst)
+  zip_xcframework_archive(dst, args)
+
+  # Generate Flutter.dSYM.zip for manual symbolification.
+  #
+  # Historically, the framework dSYM was named Flutter.dSYM, so in order to
+  # remain backward-compatible with existing instructions in docs/Crashes.md
+  # and existing tooling such as dart-lang/dart_ci, we rename back to that name
+  #
+  # TODO(cbracken): remove these archives and the upload steps once we bundle
+  # dSYMs in app archives. https://github.com/flutter/flutter/issues/153879
+  framework_dsym = framework_dst + '.dSYM'
+  if os.path.exists(framework_dsym):
+    renamed_dsym = framework_dsym.replace('FlutterMacOS.framework.dSYM', 'FlutterMacOS.dSYM')
+    os.rename(framework_dsym, renamed_dsym)
 
-  dsym_dst = os.path.join(dst, 'FlutterMacOS.dSYM')
-  if os.path.exists(dsym_dst):
     # Create a zip of just the contents of the dSYM, then create a zip of that zip.
     # TODO(cbracken): remove this once https://github.com/flutter/flutter/issues/125067 is resolved
-    sky_utils.create_zip(dsym_dst, 'FlutterMacOS.dSYM.zip', ['.'])
-    sky_utils.create_zip(dsym_dst, 'FlutterMacOS.dSYM_.zip', ['FlutterMacOS.dSYM.zip'])
+    sky_utils.create_zip(renamed_dsym, 'FlutterMacOS.dSYM.zip', ['.'])
+    sky_utils.create_zip(renamed_dsym, 'FlutterMacOS.dSYM_.zip', ['FlutterMacOS.dSYM.zip'])
 
     # Move the double-zipped FlutterMacOS.dSYM.zip to dst.
-    dsym_final_src_path = os.path.join(dsym_dst, 'FlutterMacOS.dSYM_.zip')
+    dsym_final_src_path = os.path.join(renamed_dsym, 'FlutterMacOS.dSYM_.zip')
     dsym_final_dst_path = os.path.join(dst, 'FlutterMacOS.dSYM.zip')
     shutil.move(dsym_final_src_path, dsym_final_dst_path)
 
 
-def zip_xcframework_archive(dst):
-  sky_utils.write_codesign_config(os.path.join(dst, 'entitlements.txt'), [])
-
-  sky_utils.write_codesign_config(
-      os.path.join(dst, 'without_entitlements.txt'), [
-          'FlutterMacOS.xcframework/macos-arm64_x86_64/'
-          'FlutterMacOS.framework/Versions/A/FlutterMacOS'
-      ]
-  )
-
-  sky_utils.create_zip(
-      dst,
-      'framework.zip',
-      [
-          'FlutterMacOS.xcframework',
-          'entitlements.txt',
-          'without_entitlements.txt',
-      ],
-  )
+def zip_xcframework_archive(dst, args):
+  # pylint: disable=line-too-long
+  with_entitlements = []
+  with_entitlements_file = os.path.join(dst, 'entitlements.txt')
+  sky_utils.write_codesign_config(with_entitlements_file, with_entitlements)
+
+  without_entitlements = [
```

Comment:
Can you add a comment to this and the iOS one says if this list is ever updated, also update <url to cocoon test>? That way the release team doesn't have a surprise.

Reply:
Sorry, I was writing from my phone and misremembered, the test isn't in cocoon, it's in flutter/flutter:
https://github.com/flutter/flutter/blob/62382c7b83a16b3f48dc06c19a47f6b8667005a5/dev/bots/suite_runners/run_verify_binaries_codesigned_tests.dart#L88-L108

I think that list needs to stay up-to-date with this one.  So when this PR rolls, that test should be updated to avoid release issues like https://github.com/flutter/flutter/pull/153787

Reply:
Awesome! Also on my phone but will look in a bit! Thanks for the pointer.

Reply:
Done.